packages/p2p/src/pages/advertiser-page/advertiser-page.jsx

```diff
@@ -101,28 +101,43 @@ const AdvertiserPage = () => {
 
     const setShowAdvertInfo = React.useCallback(
         () => {
-            if (p2p_advert_info) {
-                const { eligibility_status, is_active, is_buy, is_eligible, is_visible } = p2p_advert_info || {};
+            const { is_active, is_buy, is_visible } = p2p_advert_info || {};
+            if (isSubscribed && p2p_advert_info) {
                 const advert_type = is_buy ? 1 : 0;
 
-                if (is_active && is_visible && is_eligible) {
+                if (is_active && is_visible) {
                     advertiser_page_store.setActiveIndex(advert_type);
                     advertiser_page_store.handleTabItemClick(advert_type);
                     buy_sell_store.setSelectedAdState(p2p_advert_info);
                     showModal({ key: 'BuySellModal' });
                 } else {
-                    showErrorModal(eligibility_status);
+                    showErrorModal();
                 }
+            } else {
+                showErrorModal();
             }
         },
 
         // eslint-disable-next-line react-hooks/exhaustive-deps
-        [p2p_advert_info]
+        [isSubscribed, p2p_advert_info]
     );
 
     React.useEffect(() => {
-        if (is_advertiser && !is_barred && is_my_advert !== null && !is_my_advert) setShowAdvertInfo();
-    }, [counterparty_advert_id, setShowAdvertInfo, is_my_advert]);
+        if (is_advertiser && !is_barred && is_my_advert !== null && !is_my_advert) {
+            if (isLoading && isDesktop()) {
+                showModal({ key: 'LoadingModal' });
+            } else if (counterparty_advert_id) {
+                setShowAdvertInfo();
+            }
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [counterparty_advert_id, isLoading, setShowAdvertInfo, is_my_advert]);
+
+    // Update the selected advert state when the advert info is updated through subscription.
+    React.useEffect(() => {
+        if (p2p_advert_info) buy_sell_store.setSelectedAdState(p2p_advert_info);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [p2p_advert_info]);
```

Comment:
the advert info wasn't being updated since it was never setting the value in the store, so this is just to listen to the response and if there is any change, it'll update the buy sell modal accordingly. This also fixes the completion time being updated in the modal which was an issue previously for flexible order time feature